Makefile

```diff
@@ -37,6 +40,12 @@ agent: ## Build agent.
 	${MAKE} extensions
 	@ls -l agent/build
 
+agent-debug: ## Build agent for debug.
```

Comment:
this pr is pretty big, can we separate the "devops" changes which feels like to related to service map feature to a separate pr?

Reply:
#626 #624 #635